addons/l10n_in_pos/models/account_move_line.py

```diff
@@ -0,0 +1,54 @@
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _compute_l10n_in_gstr_section(self):
+        super()._compute_l10n_in_gstr_section()
+
+        def has_tags(category):
+            return any(tag in tax_tags for tag in tax_tags_ids[category])
+
+        def get_transaction_type(move):
+            return 'intra_state' if move.l10n_in_state_id == move.company_id.state_id else 'inter_state'
+
+        def is_b2cs(line):
+            return (
+                line.move_id.l10n_in_gst_treatment in ('unregistered', 'consumer') and
+                has_tags('gst') and
+                (get_transaction_type(line.move_id) == 'intra_state' or
+                (get_transaction_type(line.move_id) == 'inter_state' and
+                    (not line.move_id.reversed_entry_id and line.move_id.amount_total <= 100000) or
+                    (line.move_id.reversed_entry_id and line.move_id.reversed_entry_id.amount_total <= 100000)
+                ))
+            )
```

Comment:
As we know, it's a POS entry, so we only check for GST tags; no other conditions
POS entries should only be moved to B2CS or Nill Rated. so no complex logic here.